core/mondoo-dns-security.mql.yaml

```diff
@@ -45,44 +45,136 @@ policies:
+        * Incompatibility with Office 365: Microsoft Office 365 requires specific MX record configurations (e.g., *.mail.protection.outlook.com) to properly route and secure email traffic. Using legacy MX records may prevent Office 365 features, such as spam filtering or encryption, from functioning correctly.
+        * Administrative Overhead: Retaining legacy MX records increases complexity and the potential for mismanagement during troubleshooting or migrations.
+      remediation: |
+        Replace all legacy MX records with the correct Office 365 MX records provided in the Microsoft 365 Admin Center. Regularly audit DNS settings to ensure all MX records align with Office 365 requirements.
     refs:
       - url: https://techcommunity.microsoft.com/t5/exchange-team-blog/best-practices-for-using-assigned-office-365-dns-records/ba-p/607907
         title: Best practices for using assigned Office 365 DNS records
   - uid: mondoo-dns-security-google-workspaces-mx-records
     title: Ensure the correct MX records are used with Google Workspaces
+    impact: 80
     mql: |-
       dns.mx.where(domainName == /l.google.com/).
         map(domainName.downcase).
         containsOnly(["aspmx.l.google.com.", "alt1.aspmx.l.google.com.", "alt2.aspmx.l.google.com.", "alt3.aspmx.l.google.com.", "alt4.aspmx.l.google.com."])
+    docs:
+      desc: |
+        This security check verifies that the domain’s MX (Mail Exchange) records are correctly configured to use the Google Workspace (formerly G Suite) email servers. Proper configuration ensures reliable email delivery and leverages Google’s advanced email security and management features.
+
+        Why This Is Important:
+
+        * Reliable Email Delivery: Incorrect or missing MX records can result in undelivered emails, affecting business communication.
+        * Security Risks: Using incorrect MX records may route emails through untrusted servers, exposing the organization to spoofing, phishing, or data interception attacks.
+        * Google Workspace Functionality: Google Workspace requires specific MX records to activate its full suite of email features, including spam protection, encryption, and account-based email management.
+        * Compliance and Best Practices: Adhering to Google Workspace’s DNS configuration guidelines ensures compatibility and minimizes the risk of service disruptions.
+      remediation: |
+        Ensure the domain’s MX records point to Google’s designated email servers:
+          *	ASPMX.L.GOOGLE.COM
+          *	ALT1.ASPMX.L.GOOGLE.COM
+          *	ALT2.ASPMX.L.GOOGLE.COM
+          *	ALT3.ASPMX.L.GOOGLE.COM
+          *	ALT4.ASPMX.L.GOOGLE.COM
+
+        Verify that each record is configured with the correct priority as recommended by Google. Remove any legacy or non-Google MX records to prevent misrouting or security vulnerabilities.
```

Comment:
```suggestion
        Why this is important:

        * Reliable email delivery: Incorrect or missing MX records can result in undelivered email.
        * Security risks: Using incorrect MX records can route email through untrusted servers, exposing the organization to spoofing, phishing, or data interception attacks.
        * Google Workspace functionality: Google Workspace requires specific MX records to activate its full suite of email features, including spam protection, encryption, and account-based email management.
        * Compliance and best practices: Adhering to Google Workspace’s DNS configuration guidelines ensures compatibility and minimizes the risk of service disruptions.
      remediation: |
        Ensure the domain’s MX records point to Google’s designated email servers:
          *	ASPMX.L.GOOGLE.COM
          *	ALT1.ASPMX.L.GOOGLE.COM
          *	ALT2.ASPMX.L.GOOGLE.COM
          *	ALT3.ASPMX.L.GOOGLE.COM
          *	ALT4.ASPMX.L.GOOGLE.COM

        Verify that each record is configured with the correct priority as recommended by Google. Remove any legacy or non-Google MX records to prevent misrouting or security vulnerabilities.
```